File: packages/ai-service/app/api/health.py
"""Health check endpoints for AI Service."""

from datetime import datetime
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter()


class HealthResponse(BaseModel):
    """Health check response model."""

    status: str
    service: str
    timestamp: str
    version: str


class ReadinessResponse(BaseModel):
    """Readiness check response model."""

    status: str
    timestamp: str
    checks: dict


@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    """Basic health check endpoint."""
    return HealthResponse(
        status="healthy",
        service="ai-service",
        timestamp=datetime.utcnow().isoformat(),
        version="0.1.0",
    )


@router.get("/health/ready", response_model=ReadinessResponse)
async def readiness_check() -> ReadinessResponse:
    """Readiness check endpoint."""
    # Add checks for AI services (API keys, vector DB, etc.)
    checks = {
        "anthropic": True,  # Placeholder
        "openai": True,  # Placeholder
        "pinecone": True,  # Placeholder
    }

    is_ready = all(checks.values())

    return ReadinessResponse(
        status="ready" if is_ready else "not ready",
        timestamp=datetime.utcnow().isoformat(),
        checks=checks,
    )


@router.get("/health/live")
async def liveness_check() -> dict:
    """Liveness check endpoint."""
    return {
        "status": "alive",
        "timestamp": datetime.utcnow().isoformat(),
    }

File: packages/backend/src/routes/health.ts
import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';

export async function healthRoutes(fastify: FastifyInstance) {
  fastify.get('/health', async (_request: FastifyRequest, reply: FastifyReply) => {
    return reply.status(200).send({
      status: 'healthy',
      service: 'backend-api',
      timestamp: new Date().toISOString(),
      version: process.env.npm_package_version || '0.1.0',
      uptime: process.uptime(),
    });
  });

  fastify.get('/health/ready', async (_request: FastifyRequest, reply: FastifyReply) => {
    // Add database and other service checks here
    const isReady = true; // Placeholder

    if (isReady) {
      return reply.status(200).send({
        status: 'ready',
        timestamp: new Date().toISOString(),
      });
    } else {
      return reply.status(503).send({
        status: 'not ready',
        timestamp: new Date().toISOString(),
      });
    }
  });

  fastify.get('/health/live', async (_request: FastifyRequest, reply: FastifyReply) => {
    return reply.status(200).send({
      status: 'alive',
      timestamp: new Date().toISOString(),
    });
  });
}
